prometheus_juju_backup_all_exporter/config.py

```diff
@@ -0,0 +1,42 @@
+import os
+from enum import Enum
+from logging import getLogger
+
+from pydantic import BaseModel, validator
+from yaml import safe_load
+
+logger = getLogger(__name__)
+
+DEFAULT_CONFIG = os.path.join(os.environ.get("SNAP_DATA"), "config.yaml")
+
+
+class LogLevel(str, Enum):
+    DEBUG = "DEBUG"
+    INFO = "INFO"
+    WARNING = "WARNING"
+    ERROR = "ERROR"
+    CRITICAL = "CRITICAL"
+
+
+class Config(BaseModel):
+    port: int = 10000
+    level: LogLevel = LogLevel.DEBUG
```

Comment:
It seems the same as

```Python
logging.getLevelName(logging.DEBUG)
```

Reply:
Pydantic's `BaseModel` will enforce typing. So if you input `level=XXX`, it will report error. Whereas if you use `logging.getLevelName(XXX)` it won't.

I think you meant in `__main__.py:10`? In there, I think we can use `logging.getLevelName`

Reply:
Done

Reply:
No, I don't understand why we need to define our own log level here. It should be defined in logging library and we can use. 

If you want to store level as a enum/number, we can use
    level: int: logging.DEBUG
if you want to store level as a string, we can use
    level: str: logging.getLevelName(logging.DEBUG)

Reply:
For case one, you need to specify a level number in the `config.yaml` which is not desired.

For case two, yes, we can do something like `level: str = "DEBUG"`. But we cannot make use of `pydantic.BaseModel` to validate the input. For example, 
```python3
level = "random"
logging.getLevelName(level)
# return "Level random"
```
However, with the current approach, it will raise error. So, the custom level is used to validate log level input.

Reply:
It's used with `snap logs <snap-name>`